### DISCUSSION TITLE
This is awesome but....

### DISCUSSION THREAD

**Discussion by semi710** [nix-community/nix4nvchad#41] [General]
but i think the architecture of nvim some how limits how it is working on nix.
[nixvim](https://github.com/nix-community/nixvim) some how managed to get neovim purely under them.

Somethings from there can be adapted like how they are handling the binary and how to modularize most of the stuff.

[a comment by an automated account is not shown]

**Comment by cbaconnier** [comment]
Using the `nvchad-starter` feature from nix4nvchad, I can use my own repository for the config and it gives me the freedom for my config and binaries from nixos. Plus, nvim config can also be used outside of nixos.

This is what I have done: 
```
  nvchad-starter = {
      url = "github:cbaconnier/nvim";
      #url = "path:/home/clement/Projects/cbaconnier/nvim";   
      flake = false;
    };
  nvchad4nix = {
    url = "github:nix-community/nix4nvchad";
    inputs.nixpkgs.follows = "nixpkgs";
    inputs.nvchad-starter.follows = "nvchad-starter";
  };
  ```
binaries: https://github.com/cbaconnier/nixos-config/blob/6fc00a08087806b3e219998ad3d1a56a03712bb7/home-manager/nvim/default.nix
nvchad (really simple config at the moment): https://github.com/cbaconnier/nvim

So far, my pain point is testing / changing my nvim configuration. My current workflow is;
- Update the nvchad-starter url to my local path
- Do the changes in nvchad - nvim
- Rebuild nixos 
- Repeat
- Once I'm done, commit nvchad - nvim
- Updating nvchad-starter url with the repository
- Commit
- :face_exhaling: 

I will probably use a VM to configure nvim because in the end I feel like it will be easier this way. 
But this workflow is most probably a lack of knowledge from me.

**Reply by MOIS3Y** [reply]
Hey @cbaconnier, if you’re adding your NvChad configuration as a flake input, you can manage the config in pure Lua much more easily.

Here’s how I do it:

- Wipe the current config completely:
 ```console
 sudo rm -rf ~/.config/nvim* ~/.local/share/nvim ~/.local/state/nvim ~/.cache/nvim/  
 ```
- Symlink my local repo:
I keep a separate repo with my NvChad config in pure Lua:
```console
ln -s ~/Documents/Dev/Self/nvchad-on-steroids ~/.config/nvim  
```
- Tweak, extend, and test the config.
- If everything works well, commit the changes and push to the remote repo.
- Clean up NvChad artifacts again.
- To update the input for my NixOS config, I refresh it in the flake:
```console
nix flake update nvchad-on-steroids  
```
- Then run:
```console
sudo nixos-rebuild switch
```

Bonus note:
You don’t even need to extend `extraPackages` in your NixOS config. Most of these packages are for LSP, but since LSP loads lazily, you can always use `nix shel`l or `nix develop` to create an environment with the required LSP servers when need
